src/modules/SdnDiag.Health.psm1

```diff
@@ -2,1467 +2,1638 @@
 
-        $ncNodes = Get-SdnServiceFabricNode -NetworkController $SdnEnvironmentObject.ComputerName[0] -Credential $credential
-        if($null -eq $ncNodes){
-            throw New-Object System.NullReferenceException("Unable to retrieve service fabric nodes")
-        }
-
-        $applicationHealth = Get-SdnServiceFabricApplicationHealth -NetworkController $SdnEnvironmentObject.ComputerName[0] -Credential $Credential
-        if ($applicationHealth.AggregatedHealthState -ine 'Ok') {
-            $sdnHealthObject.Result = 'FAIL'
-            $sdnHealthObject.Remediation += "Examine the Service Fabric Application Health for Network Controller to determine why the health is not OK."
-        }
-
-        return $sdnHealthObject
-    }
-    catch {
-        $_ | Trace-Exception
-        $_ | Write-Error
-    }
-}
+        .PARAMETER ApiVersion
+        (optional) The version of the resource to retrieve from the Network Controller.
+        note: if nothing is specified, v1 is queried
 
-function Test-ServiceFabricClusterHealth {
-    <#
-    .SYNOPSIS
-        Validate the health of the Network Controller cluster within Service Fabric.
     #>
 
     [CmdletBinding()]
     param (
+        [Parameter(Mandatory = $false)]
+        [string] $NcUri,
+
         [Parameter(Mandatory = $true)]
-        [SdnFabricEnvObject]$SdnEnvironmentObject,
+        [ValidateSet(
+                'Servers', `
+                'NetworkInterfaces', `
+                'VirtualNetworks', `
+                'LogicalNetworks'
+        )]
+        [String]$ResourceType,
 
         [Parameter(Mandatory = $false)]
-        [System.Management.Automation.PSCredential]
-        [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty
+        [String]$ApiVersion = 'v1'
     )
-
-    $sdnHealthObject = [SdnHealth]::new()
-
     try {
-        "Validating the Service Fabric Cluster Health for Network Controller" | Trace-Output
-
-        $ncNodes = Get-SdnServiceFabricNode -NetworkController $SdnEnvironmentObject.ComputerName[0] -Credential $credential
-        if($null -eq $ncNodes){
-            throw New-Object System.NullReferenceException("Unable to retrieve service fabric nodes")
+        $NcUri = $NcUri.TrimEnd('/')
+        # kerberos is not supported yet
+        $resources = $null
+        # try all certificates from machine & user store
+        $certName = Get-ItemProperty -Path HKLM:\SYSTEM\CurrentControlSet\Services\NcHostAgent\Parameters\ -Name HostAgentCertificateCName
+        $certName = $certName.HostAgentCertificateCName
+        $certs = Get-ChildItem -Path Cert:\LocalMachine\My | ?{ $_.Subject -eq "CN=$certName" }
+        $certs += Get-ChildItem -Path Cert:\CurrentUser\My | ?{ $_.Subject -eq "CN=$certName" }
+        # also try without a client cert
+        $certs += $null
+        [System.Array]::Reverse($certs)
+
+        foreach ($cert in $certs) {
+            if($null -eq $cert) {
```

Comment:
@sbgms we should move this $null check outside of the loop. If was $null, would not enter the foreach loop.

You can also leverage `Get-SdnServerCertificate` which would return back the requested certificate. It's within the `SdnDiag.Server.psm1` file, and is exported function available when SdnDiagnostics module is imported.

Reply:
Thanks, I've taken a dep on Get-SdnServerCertificate. 

As for the null check, its intentional. the check is on $cert and not $certs. Its meant to try out no-certificate scenario, when there is no client authentication specified.

Reply:
Btw, FYI Get-SdnServerCertificate returns multiple certificates in AZ local environments.

Reply:
@sbgms ya I figured that might be the case. I have created PR #369 to improve the cert cmdlets to locate the certificates with the NC OID. I will be testing on AZ Local today.